envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/listeners/EnvoyListenersFactory.kt

```diff
@@ -105,21 +115,52 @@ class EnvoyListenersFactory(
                                         .setAddress(listenersConfig.ingressHost)
                         )
                 )
-                .addFilterChains(createIngressFilterChain(group, listenersConfig, globalSnapshot))
-                .build()
+
+        listOfNotNull(securedIngressChain, insecureIngressChain).forEach {
+            listener.addFilterChains(it.build())
+        }
+
+        return listener.build()
     }
 
     private fun createIngressFilterChain(
         group: Group,
         listenersConfig: ListenersConfig,
-        globalSnapshot: GlobalSnapshot
-    ): FilterChain {
-
-        val filterChain = FilterChain.newBuilder()
+        globalSnapshot: GlobalSnapshot,
+        transportProtocol: String = "raw_buffer"
+    ): FilterChain.Builder {
+        return FilterChain.newBuilder()
+                .setFilterChainMatch(FilterChainMatch.newBuilder().setTransportProtocol(transportProtocol))
                 .addFilters(createIngressFilter(group, listenersConfig, globalSnapshot))
```

Comment:
I think we should differentiate metrics for raw_buffer chain and tls chain. So we should change `statPrefix` of http_connection_manager at least.